### ISSUE TITLE
VRF outputs and multiplied floor

### ISSUE THREAD

**Issue by BobHenninger** [NREL/EnergyPlus#4402]
Helpdesk Ticket 9656

I recall some e-mail exchange with Rich about this where I declared this was probably a bug, but I don't see an issue posted yet.
 Michael J. Witte on 12 August 2014 09:35 PM

Posted on: 18 July 2014 06:54 PM
Your description sounds very familiar to something I heard a while ago. 
 I am checking to see if there is documentation or something else that needs updating.
 'Richard Raustad

Thanks Richard,
 I tested using autosized inputs and I compared results.
 To the simulation runs OK I must multiply the cooling coil capacities (both in zone and system), but must not multiply the air flow. 
 I look at all the syzing summaries available: component, equipment and HVAC. In this case, especially the VRF component summary outputs.
 Tks a lot,
 Thais Inatomi

You might test using autosized inputs to see what happens.
 I also suggest sending us the 2 input files with only the 1 change you mentioned.
 This way we know what you have entered as inputs.
 Also tell us exactly which table reports you are looking at.
 'Richard Raustad

Hi,

 I’m modeling VRF system with actual designed capacities and actual air flows. 
 The model has one floor multiplied by 8 (I’m using zone list/zone group for that).
 I’m having problems with the output table considering this multiplied floor.

 Should I input capacities and flows from VRF units multiplied by 8 in the VRF zones and systems? Or not?
 When I do not multiply it, I have a lot of unmet load hours. However, the table presents all the flows and capacities multiplied automaticaly by 8. That would be OK if the unmet load hours weren’t so many.
 When I put the multiplied zone list/group to only 1, the unmet load hours completly disappear.
 When I input all the capacities and flows multiplied by 8, the table results are clearly wrong (not only the equipments sizes but also the energy consumption from fans and cooling coils).

 I am understanding that maybe there is a bug with the account of unmet load hours output. 
 Could you help me?

 thanks a lot.
 Thais Inatomi

User files
Although idf files were requested from the user, it appears that they were never sent

**Comment by rraustad** [comment]
Ticket 9656 has an input file attached. I added that file to defect files in EnergyPlusDevSupport. I just can't recall where the zone multipliers come into play. Before or after the zone air balance? I think I remember that the zone coils have airflow and capacity needed for the unmultiplied zone load but I'm not sure.

**Comment by mjwitte** [comment]
@rraustad To be consistent with other HVAC equipment, the zone coils need to have airflow and capacity that match the _multiplied_ zone load.

**Comment by rraustad** [comment]
I see this in HVACStandAloneERV at line 1257 where the FinalZoneSizing data is NOT used and air flow is calculated based on zone floor area. I guess from this I assume the FinalZoneSizing data DOES HAVE the multiplier included in airflow numbers.

ZoneMult = Zone( ActualZoneNum ).Multiplier \* Zone( ActualZoneNum ).ListMultiplier;
.snip.
SupplyAirVolFlowDes = ZoneMult \* SupplyAirVolFlowDes;

I'll have to check to see if this needs to be done for VRF or if it's something else happening here. Certainly for UnitarySystem, when alternate sizing is used (e.g. based on floor area), this needs to be done there as well.

**Comment by mjwitte** [comment]
@rraustad HVACStandAloneERV is different, because it "autosizes" based on inputs that are directly in the object - doesn't really fit the mold of other zone equipment when it comes to sizing.

**Comment by rraustad** [comment]
I think the user was confused here. The defect file was only a partial snapshot and would not run. The Zone List did not point to actual zones, or those zones were not included in the submitted file.
    ZoneList,
       4P,                      !- Name
       4P_ZN01,                 !- Zone 1 Name
       4P_ZN02,                 !- Zone 2 Name
       4P_ZN03,                 !- Zone 3 Name

What I believe should happen, is the zone multipliers get applied in the background and the equipment models see a multiplied zone air flow rate. The results of the following comparison support that "supposition".

The VariableRefrigerantFlow_5Zone example file was modified as:

```
Zone,
 SPACE1-1,                !- Name
 0,                       !- Direction of Relative North {deg}
 0,                       !- X Origin {m}
 0,                       !- Y Origin {m}
 0,                       !- Z Origin {m}
 1,                       !- Type
 1,                       !- Multiplier
 2.438400269,             !- Ceiling Height {m}
 239.247360229;           !- Volume {m3}

ZoneGroup,
 SPACE1-1 Zone Group,               !- Name
 SPACE1-1 List,          !- Zone List Name
 10;                       !- Zone List Multiplier

ZoneList,
 SPACE1-1 List,          !- Name
 PLENUM-1,
 SPACE1-1,           !- Zone 3 Name
 SPACE2-1,           !- Zone 3 Name
 SPACE3-1,           !- Zone 3 Name
 SPACE4-1,           !- Zone 3 Name
 SPACE5-1;           !- Zone 3 Name
```

All zones have a multiplier of 10. This multiplied file was compare to the original example file. Results are as expected in most cases. I did see some "not multiplied by 10" results which I will track down next.

Results for the original example file:
![vrf_componentsummary_x1](https://cloud.githubusercontent.com/assets/5778156/9833749/a6178a1a-5971-11e5-8b7b-a0ea24bccbaf.png)

Results for multiplied file where spaces were x10:
![vrf_componentsummary_x10](https://cloud.githubusercontent.com/assets/5778156/9833750/ad8de8fc-5971-11e5-8840-5bca9430616d.png)

**Comment by rraustad** [comment]
Why would the normalized metrics change? Whether a zone is not or is multiplied should not change the result. I guess if SPACE1-1 were not equal to the average then this result could change. This is the total building so maybe this zone is dominant.

![normailzedmetrics](https://cloud.githubusercontent.com/assets/5778156/9778000/b0e03b4e-5738-11e5-8ff2-6edacdc3350f.png)

**Comment by rraustad** [comment]
Lighting appears to be correct. Left column x1 and right x10:

![lighting](https://cloud.githubusercontent.com/assets/5778156/9778076/a9ded02a-5739-11e5-8430-81de010d48ea.png)

**Comment by mjwitte** [comment]
@rraustad I would just 10x all the zones to make it simple to compare.

**Comment by rraustad** [comment]
Interesting result. Top 2 tables, average and minimum, are x1, bottom 2 are x10.

![oa](https://cloud.githubusercontent.com/assets/5778156/9778251/f888ac80-573b-11e5-8065-a48ec3054e72.png)

**Comment by rraustad** [comment]
OK, I agree. There does seem to be something wrong here, but it's not in the equipment models.

**Comment by rraustad** [comment]
HVAC Sizing Summary also appears correct in all places where multipliers are expected to affect results.

Original file:
![vrf_hvacsizingsummary_x1](https://cloud.githubusercontent.com/assets/5778156/9833789/c1154aee-5973-11e5-9e2a-3c7abfbcbaff.png)

Multiplied x10 file:
![vrf_hvacsizingsummary_x10](https://cloud.githubusercontent.com/assets/5778156/9833790/c5c71752-5973-11e5-9069-e7a2775dc7e0.png)

**Comment by rraustad** [comment]
The outdoor air summary does show an issue since this table is supposed to be non-multiplied results. This is a little problematic since the programmer has a difficult time with when variables are multiplied and when they are not. Additionally, ACH should not have changed regardless of multipliers.

Original file:
![vrf_outdoorairsummary_x1](https://cloud.githubusercontent.com/assets/5778156/9833919/02511938-5975-11e5-90ba-6fb2985e961f.png)

Multiplied x10 file:
![vrf_outdoorairsummary_x10](https://cloud.githubusercontent.com/assets/5778156/9833920/0656fa98-5975-11e5-80b2-43b4c3321298.png)

**Comment by rraustad** [comment]
Definitely some funky math going on in the Sensible Heat Gain Summary Report.

Original file:
![vrf_sensibleheatgainsummary_x1](https://cloud.githubusercontent.com/assets/5778156/9833993/f35e2e7c-5977-11e5-97b8-4f643c22297d.png)

Multiplied x10 file:
![vrf_sensibleheatgainsummary_x10](https://cloud.githubusercontent.com/assets/5778156/9833996/f8e2b16a-5977-11e5-8749-c7308fa84488.png)

**Comment by rraustad** [comment]
Lighting Summary shows correct except for consumption, last column.

Original file:
![vrf_lightingsummary_x1](https://cloud.githubusercontent.com/assets/5778156/9834037/f36a7f78-5978-11e5-9790-f419c75bd56d.png)

Multiplied x10 file:
![vrf_lightingsummary_x10](https://cloud.githubusercontent.com/assets/5778156/9834039/f99499ec-5978-11e5-8bf4-d2b24b6149d2.png)

**Comment by rraustad** [comment]
There appears to be issues with zone list and/or group multipliers. The places in code where multipliers are used include both types, so the problem lies in where these multipliers "should" be applied, versus where they have been applied (correctly or incorrectly).

ZoneMult = Zone( ZoneNum ).Multiplier \* Zone( ZoneNum ).ListMultiplier;

**Comment by rraustad** [comment]
This next comparison was not as apparent to where the problems where until this exercise (and still not all inclusive). Since there are only some places where multipliers are not used, the results are relatively close. These results should be exactly x10 for column 1, but are not. The last 2 columns show that the per area (multiplied area IS correct) values are also off since there are some places where multipliers are not included. Additionally, anyone fixing this problem that does not include other reports (i.e., any 0.000 in the above figures) will most likely miss something.

Original file:
![vrf_abups_x1](https://cloud.githubusercontent.com/assets/5778156/9834105/959bde80-597a-11e5-9544-b63eb97c221d.png)

Multiplied x10 file:
![vrf_abups_x10](https://cloud.githubusercontent.com/assets/5778156/9834107/99c9109a-597a-11e5-8f3f-4515706d53a8.png)

**Comment by rraustad** [comment]
The Peak Cooling Sensible Heat Gain Summary:

Original file:
![vrf_peakcoolingsensibleheatgaincomponents_x1](https://cloud.githubusercontent.com/assets/5778156/9834787/f4abf374-5998-11e5-8bfc-48ec243712da.png)

Modified x10 file:
![vrf_peakcoolingsensibleheatgaincomponents_x10](https://cloud.githubusercontent.com/assets/5778156/9834789/f8f3a79c-5998-11e5-9795-0d353e47dddf.png)

**Comment by rraustad** [comment]
This is obviously confusing, since the zone air flow was multiplied correctly, even though the zone lighting, equipment, and people were not. So that raises the question of when the multiplier should be applied. It seems that the non-multiplied zone load components are used to calculate a total non-multiplied zone load, and then at some later point in the simulation, the multipliers are used to calculate the total energy. The tables shown here need to carefully account for the multipliers.

**Comment by rraustad** [comment]
This type of calculation is used for table data:

mult = Zone( zonePt ).Multiplier \* Zone( zonePt ).ListMultiplier;
sumArea += Zone( zonePt ).FloorArea \* mult;
sumPower += Lights( Loop ).DesignLevel \* mult;
PreDefTableEntry( pdchInLtArea, liteName, Zone( zonePt ).FloorArea \* mult );
PreDefTableEntry( pdchInLtPower, liteName, Lights( Loop ).DesignLevel \* mult );
PreDefTableEntry( pdchInLtDens, "Interior Lighting Total", sumPower / sumArea, 4 );
PreDefTableEntry( pdchInLtArea, "Interior Lighting Total", sumArea );
PreDefTableEntry( pdchInLtPower, "Interior Lighting Total", sumPower );

**Comment by rraustad** [comment]
@JasonGlazer These 2 equations (line 5664 OutputReportTabular) are different and it looks like the 2nd is missing ZonePreDefRep( iZone ).TotTimeOcc in the denominator AND parenthesis around the denominator multiplication. Why are the multipliers used in these equations? I don't see where ZonePreDefRep( iZone ).MechVentVolTotal is adjusted by the multipliers. See the Outdoor Air Summary comparison above where the minimum OA during occupied hours ACH is a factor of 100 different. Since this is ACH, why should the magnitude change when multipliers are used? The footnote also says "without multipliers".

//Mechanical ventilation
if ( ZonePreDefRep( iZone ).TotTimeOcc > 0 ) {
PreDefTableEntry( pdchOaoAvgMechVent, Zone( iZone ).Name, ZonePreDefRep( iZone ).MechVentVolTotal / ( ZonePreDefRep( iZone ).TotTimeOcc \* Zone( iZone ).Volume \* Zone( iZone ).Multiplier \* Zone( iZone ).ListMultiplier ), 3 );
}
if ( ( Zone( iZone ).Volume > 0 ) && ( ZonePreDefRep( iZone ).TotTimeOcc > 0 ) ) {
PreDefTableEntry( pdchOaoMinMechVent, Zone( iZone ).Name, ZonePreDefRep( iZone ).MechVentVolMin / Zone( iZone ).Volume \* Zone( iZone ).Multiplier \* Zone( iZone ).ListMultiplier, 3 );
                    }

**Comment by mjwitte** [comment]
@rraustad MechVentVolTotal and MechVentVoMin are on the HVAC system side, so they are already multiplied to start with.  I agree it looks like the denominator for Min OA is wrong as you say.  But not 100% certain.

**Comment by rraustad** [comment]
OK, I see where the multipliers would be included in the results above. The OA mass flow rate should already include the impact of multipliers.

AirSysOutAirFlow = Node( OutAirNode ).MassFlowRate;

I also see where the Min value is just that, a min. The total is a sum over time and the denominator would include TotTimeOcc and Vol. The min calc would only include Vol.

**Comment by rraustad** [comment]
OA Summary now looks like this. Also fixed the Consumption value (x10) in the Lighting Summary Report.

![vrf_oasummary_modified_new](https://cloud.githubusercontent.com/assets/5778156/9838645/9a2a0556-5a30-11e5-85ea-db2e62f23ee6.png)

**Comment by rraustad** [comment]
The meters look like they should honor the multipliers. In UpdateMeterValues, the meters should all be adjusted by the zone and list multipliers.
UpdateMeterValues( rVar.TSValue \* rVar.ZoneMult \* rVar.ZoneListMult, VarMeterArrays( rVar.MeterArrayPtr ).NumOnMeters, VarMeterArrays( rVar.MeterArrayPtr ).OnMeters );

So I added the following meters to each file to see if the x10 worked for all meters:
    Output:Meter,Electricity:Facility,runperiod; !- [J]
    Output:Meter,Electricity:Building,runperiod; !- [J]
    Output:Meter,Electricity:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,GeneralLights:InteriorLights:Electricity,runperiod; !- [J]
    Output:Meter,Electricity:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,Electricity:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,Electricity:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,Electricity:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,InteriorLights:Electricity:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,General:InteriorEquipment:Electricity,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,InteriorEquipment:Electricity:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Facility,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Building,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:PLENUM-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:PLENUM-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:PLENUM-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:SPACE1-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:SPACE2-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:SPACE3-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:SPACE4-1,runperiod; !- [J]
    Output:Meter,EnergyTransfer:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,Heating:EnergyTransfer:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,Cooling:EnergyTransfer:Zone:SPACE5-1,runperiod; !- [J]
    Output:Meter,ElectricityPurchased:Facility,runperiod; !- [J]
    Output:Meter,ElectricityPurchased:Plant,runperiod; !- [J]
    Output:Meter,Cogeneration:ElectricityPurchased,runperiod; !- [J]
    Output:Meter,ElectricityNet:Facility,runperiod; !- [J]
    Output:Meter,ElectricityNet:Plant,runperiod; !- [J]
    Output:Meter,Cogeneration:ElectricityNet,runperiod; !- [J]
    Output:Meter,Electricity:HVAC,runperiod; !- [J]
    Output:Meter,Fans:Electricity,runperiod; !- [J]
    Output:Meter,General:Fans:Electricity,runperiod; !- [J]
    Output:Meter,EnergyTransfer:HVAC,runperiod; !- [J]
    Output:Meter,CoolingCoils:EnergyTransfer,runperiod; !- [J]
    Output:Meter,HeatingCoils:EnergyTransfer,runperiod; !- [J]
    Output:Meter,Cooling:Electricity,runperiod; !- [J]
    Output:Meter,Heating:Electricity,runperiod; !- [J]
    Output:Meter,Carbon Equivalent:Facility,runperiod; !- [kg]
    Output:Meter,CarbonEquivalentEmissions:Carbon Equivalent,runperiod; !- [kg]

Most of these meters show an exact x10 multiplier. The following shows those meters that did not have an exact x10 multiple. The values shown are for the original file, the next number (those near 10) are the division of the x10 file by the original file. Not sure yet how to track these down.

Cooling RunPeriod:
![vrf_metercompare](https://cloud.githubusercontent.com/assets/5778156/9838789/75ebf7fa-5a33-11e5-8980-35af915bc683.png)

Heating RunPeriod:
![vrf_metercompare_heating](https://cloud.githubusercontent.com/assets/5778156/9838812/53b798e6-5a34-11e5-9f6e-9a6cf5d42845.png)

**Comment by mjwitte** [comment]
@rraustad Is all of the HVAC autosized?  And are all the sizes reported to eio 10x?  Are there any exterior energy uses?

**Comment by rraustad** [comment]
I'll get to the eio next. I don't see any exterior energy uses in VariableRefrigerantFlow_5Zone. Could you  also check?

I repeated this exercise using a minimum time step of 15 minutes to eliminate any issues with reduced time steps. The answer did change. The following figure is the cooling run period followed by the heating run period.

```
ConvergenceLimits,
 15,                       !- Minimum System Timestep {minutes}
 25;                      !- Maximum HVAC Iterations
```

![vrf_metercompare_min15mints](https://cloud.githubusercontent.com/assets/5778156/9838876/9f2f8fb6-5a36-11e5-9afd-8cc14a210e59.png)

**Comment by mjwitte** [comment]
@rraustad If the eio sizes don't change by 10x, then the meters won't either.  So, that has to be right first.

**Comment by rraustad** [comment]
eio sizes are very nearly x10. Warmup convergence showed slight differences. Given that an exact x10 was not achieved and warmup differences exist, expecting results to be exactly x10 is not expected. So only gross differences will be tracked down (i.e., html diff's that are obvious) for this issue.

eio air flow comparison (x10 eio divided by original eio):

![vrf_eio_airflow](https://cloud.githubusercontent.com/assets/5778156/9839257/adc80282-5a41-11e5-8477-5868b2626361.png)

Sizing results:

![vrf_eio_sizing](https://cloud.githubusercontent.com/assets/5778156/9839259/b66fd586-5a41-11e5-8157-428fc51df205.png)

**Comment by rraustad** [comment]
Two of the issues have been corrected:
1) Outdoor Air Summary Report: Mechanical Ventilation for Minimum Outdoor Air During Occupied Hours
2) Lighting Summary Report: Consumption value (x10)

For the Sensible Heat Gain Summary Report, since 2 of the columns are differences of the remaining columns, these values must all be multiplied, or not multiplied. I will correct these values assuming all need to be multiplied values.

Any other comments or suggestions?

**Comment by rraustad** [comment]
New Sensible Heat Gain Summary Reports:

![vrf_componentsummary_x1_newcompare](https://cloud.githubusercontent.com/assets/5778156/9839842/b7afedda-5a52-11e5-8406-fd5397bd2d98.png)

![vrf_clgcomponentsummary_x1_newcompare](https://cloud.githubusercontent.com/assets/5778156/9839848/c4270f26-5a52-11e5-8c9c-8c4f399c9e3a.png)

![vrf_htgcomponentsummary_x1_newcompare](https://cloud.githubusercontent.com/assets/5778156/9839852/ce0bc7a2-5a52-11e5-82a0-58ec8c460800.png)

**Comment by JasonGlazer** [comment]
@rraustad It looks like you figured out what you needed to regarding the OA ventilation report for mechanical ventilation. Let me know if you need anything else from me.

**Comment by rraustad** [comment]
This result looks better, but some of the columns don't add up. I guess that's a result of how cooling and heating peak are determined, and that some of these variables could be positive or negative during a cooling or heating peak (e.g., Interzone Air Transfer).

![vrf_componentsummary_x1_newcompare_final](https://cloud.githubusercontent.com/assets/5778156/9865908/6af5f470-5b2a-11e5-9e64-30a4e2451a01.png)

![vrf_clgcomponentsummary_x1_newcompare_final](https://cloud.githubusercontent.com/assets/5778156/9865913/777cf3d8-5b2a-11e5-8397-ef603a57bade.png)

![vrf_htgcomponentsummary_x1_newcompare_final](https://cloud.githubusercontent.com/assets/5778156/9865921/84fe761c-5b2a-11e5-9898-de737960486b.png)

**Comment by rraustad** [comment]
@mbadams5 
How to add clear_state to an external function?

void
CalcHeatBalanceInsideSurf( Optional_int_const ZoneToResimulate = _ ); // if passed in, then only calculate surfaces that have this zone

**Comment by mbadams5** [comment]
@rraustad What exactly are you trying to clear? The namespace in that file already has a clear_state. If you are trying to clear a function static variable then that is a different problem.

**Comment by rraustad** [comment]
Can I do this?
    void
    clear_state()
    {
        ManageSurfaceHeatBalancefirstTime = true;
        InitSurfaceHeatBalancefirstTime = true;
        ComputeIntSWAbsorpFactorsfirstTime = true;
        UpdateThermalHistoriesFirstTimeFlag = true;
        CalculateZoneMRTfirstTime = true;
        calcHeatBalanceInsideSurfFirstTime = true;

```
            CalcHeatBalanceInsideSurf::TempInsOld.deallocate();
            CalcHeatBalanceInsideSurf::RefAirTemp.deallocate();
            CalcHeatBalanceInsideSurf::calcHeatBalanceInsideSurfFirstTime = true;
}
```

**Comment by rraustad** [comment]
Wait a minute, calcHeatBalanceInsideSurfFirstTime is already in the clear state. I just need to add the other variables I want cleared.

So I just add these and any others necessary.
TempInsOld.deallocate();
RefAirTemp.deallocate();

**Comment by mbadams5** [comment]
It won't work directly. You need to pull out the needed function static variables into [here](https://github.com/NREL/EnergyPlus/blob/be934ee1c701c5ee563f65804975aa42070da112/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L132). Then you can call deallocate() on those variables in the clear_state directly. 

But looking at the code, when calcHeatBalanceInsideSurfFirstTime is reset to true it will reallocate those two arrays [here](https://github.com/NREL/EnergyPlus/blob/be934ee1c701c5ee563f65804975aa42070da112/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L4985). Try to only pull out static variables that are needed to make the unit test(s) function.

**Comment by rraustad** [comment]
A previous unit test has 6 surfaces, I have 12. Those 2 arrays were still allocated to 6. Will it reallocate those two arrays if a previous unit test already allocated them? I did move those variables last night but got a compile error. It was 1 AM so maybe I made a mistake.

Now that I see what's happening, maybe the result will be different tonight.

**Comment by mbadams5** [comment]
Looking at the code it should. But if the other test doesn't use EnergyPlusFixture then maybe not. I can't say for certain without playing around with the code.

**Comment by rraustad** [comment]
Ah, good point. But that should have only been a problem for the first unit test within EnergyPlusFixture (unless clear-state is ran "before" the first unit test).

I'll figure something out tonight to get my unit test running. 
Thanks.

**Comment by EnergyArchmage** [comment]
I had trouble before with RefAirTemp in unit tests and made all those changes to clear_state and calcHeatBalanceInsideSurfFirstTime.  I assume c++ will reallocate an array that is already allocated and it seemed to be working for me as is.

**Comment by mjwitte** [comment]
@rraustad Sensible Heat Gain report columns should add to zero.  If they don't, then something is broken.

**Comment by Myoldmopar** [comment]
Closed via #5192
